sdk/src/main/scala/io/horizen/account/state/StateTransition.scala

```diff
@@ -146,4 +128,55 @@ class StateTransition(
     // return remaining gas to the gasPool of the current block so it is available for the next transaction
     blockGasPool.addGas(gas.getGas)
   }
+
+  @throws(classOf[InvalidMessageException])
+  @throws(classOf[ExecutionFailedException])
+  def execute(invocation: Invocation): Array[Byte] = {
+    // limit call depth to 1024
+    if (invocationStack.length >= 1024) throw new ExecutionRevertedException("Maximum depth of call stack reached")
```

Comment:
Reading from the PR specs, this stack depth value taken from the current geth implementation.
If the default in the EVM is to allocate 63/64th of the currently available gas to the nested call and if we consider the current block gas limit this call stack will not be reached, right? The tx gas will play the upper limit in this case.

Reply:
Possible, yes. It all depends on the gas parameters, how much gas we consume for what, etc.